doc/application/index.rst

```diff
@@ -299,19 +294,34 @@ The Zephyr build system compiles and links all components of an application
 into a single application image that can be run on simulated hardware or real
 hardware.
 
-As described in :ref:`getting_started_cmake`, on Linux and macOS you can choose
-between the ``make`` and ``ninja`` generators, whereas on Windows you need to use
-``ninja``. For simplicity we will use ``ninja`` throughout this guide.
+Like with any other CMake-based system, the build process takes place :ref:`in
+two stages <cmake-details>`. First build files (also known as buildsystem) are
+generated using the ``cmake`` command-line tool while specifying a generator.
+This generator determines which native build tool to generate a buildsystem for.
+The second stage consists on running the native build tool to actually build the
+source files and generate an image. To learn more about these concepts refer to
+the `CMake introduction`_ in the official CMake documentation.
+
+As described in :ref:`getting_started_cmake`, you can choose to invoke ``cmake``
+directly or to use :ref:`west <west>`, Zephyr's meta-tool, which itself invokes
+``cmake`` and the build tool behind the scenes.
+On Linux and macOS you can choose between the ``make`` and ``ninja``
```

Comment:
Just a random question, not asking for a change, but I notice you put single newlines after sentences sometimes in the documentation. If you don't mean to create separate paragraphs, can I ask why, since they won't render that way in the HTML or PDF?

Reply:
Completely unintentional, probably just eye candy for me while editing the file.

Reply:
In ReST, paragraphs are determined by a blank line, so ending a sentence with a newline and starting the next sentence on the next line (with no blank line between them) will still render the two sentences in the same paragraph.